### PR TITLE
Move onboarding screen to a new Activity

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://www.pivotaltracker.com/story/show/169253555

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/di/AppComponent.kt [hunk 1]
```diff
@@ -3,14 +3,13 @@ package org.simple.clinic.di
...
 import org.simple.clinic.ReleaseClinicApp
-import org.simple.clinic.activity.TheActivityComponent
+import org.simple.clinic.main.TheActivityComponent
 import org.simple.clinic.facility.FacilitySync
 import org.simple.clinic.login.OtpSmsReceiver
 import org.simple.clinic.newentry.clearbutton.ClearFieldImageButton
 import org.simple.clinic.storage.Migration_27_28
 import org.simple.clinic.storage.Migration_29_30
 import org.simple.clinic.storage.Migration_34_35
-import org.simple.clinic.storage.Migration_48_49
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/di/AppComponent.kt]: Curious, what happened to this migration?

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -84,6 +84,10 @@
       android:theme="@style/Clinic.V2.Theme.BottomSheetActivity"
       android:windowSoftInputMode="adjustResize" />
 
+    <activity
+      android:name=".setup.SetupActivity"
+      android:enabled="false"/>
```
Comment (ragunathjawahar) [inline app/src/main/AndroidManifest.xml]: Curious, what does `android:enabled="false"` do?

File: app/src/main/java/org/simple/clinic/setup/SetupActivity.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.simple.clinic.BuildConfig
+import org.simple.clinic.ClinicApp
+import org.simple.clinic.util.LocaleOverrideContextWrapper
+import java.util.Locale
+import javax.inject.Inject
+
+class SetupActivity : AppCompatActivity() {
+
+  companion object {
+    lateinit var component: SetupActivityComponent
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/setup/SetupActivity.kt]: Hmmm... not really comfortable doing this :(

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/main/AndroidManifest.xml]: The activity cannot be started unless it is manually enabled. Basically meant to stop someone from accidentally starting it.

File: app/src/main/java/org/simple/clinic/setup/SetupActivity.kt (hunk 3 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/setup/SetupActivity.kt]: I have made it a private property in a later commit. This was initially copied from the `TheActivity`.

File: app/src/main/java/org/simple/clinic/setup/SetupActivity.kt [hunk 4]
```diff
@@ -44,7 +49,12 @@ class SetupActivity : AppCompatActivity() {
   override fun attachBaseContext(baseContext: Context) {
     setupDiGraph()
     val contextWithOverriddenLocale = LocaleOverrideContextWrapper.wrap(baseContext, locale)
-    super.attachBaseContext(ViewPumpContextWrapper.wrap(contextWithOverriddenLocale))
+    val contextWithRouter = wrapContextWithRouter(contextWithOverriddenLocale)
+    val contextWithInjectorProvider = InjectorProviderContextWrapper.wrap(
+        contextWithRouter,
+        mapOf(OnboardingScreenInjector::class.java to component)
+    )
+    super.attachBaseContext(ViewPumpContextWrapper.wrap(contextWithInjectorProvider))
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/setup/SetupActivity.kt]: The result of a wrap is fed into the next call. It's easy to make a mistake here. Can we do something like this?

```kotlin
  override fun attachBaseContext(baseContext: Context) {
    setupDiGraph()
    val wrappedContext = baseContext
        .wrap { LocaleOverrideContextWrapper.wrap(it, locale) }
        .wrap { wrapContextWithRouter(it) }
        .wrap { InjectorProviderContextWrapper.wrap(it, mapOf(OnboardingScreenInjector::class.java to component)) }
        .wrap { ViewPumpContextWrapper.wrap(it) }
    super.attachBaseContext(wrappedContext)
  }

  private fun Context.wrap(wrapper: (Context) -> Context): Context =
      wrapper(this)
```

File: app/src/main/java/org/simple/clinic/di/AppComponent.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/di/AppComponent.kt]: This was an unused import. I think a larger migration was broken into two, and the injection moved to the later commit, but the original import was left.

File: app/src/main/java/org/simple/clinic/setup/SetupActivity.kt (hunk 4 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/setup/SetupActivity.kt]: Looks good, I'll do this in #906 since that class has already changed in that PR.

Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/setup/SetupActivity.kt]: Done.

**Pull request merged**